examples/mnist/tf_example.py

```diff
@@ -0,0 +1,137 @@
+
+###
+# Adapted to petastorm dataset using original contents from
+# https://github.com/tensorflow/tensorflow/blob/master/tensorflow/examples/tutorials/mnist/mnist_softmax.py
+###
+
+from __future__ import division, print_function
+
+import argparse
+import os
+import tensorflow as tf
+
+from examples.mnist import DEFAULT_MNIST_DATA_PATH
+from petastorm.reader import Reader
+from petastorm.tf_utils import tf_tensors
+
+
+def train_and_test(dataset_url, epochs, batch_size, log_interval):
+    """
+    Train a model for epochs with a batch size batch_size, printing accuracy every log_interval.
+    :param dataset_url: The MNIST dataset url.
+    :param epochs: The epochs to train for.
+    :param batch_size: The batch size for training.
+    :param log_interval: The interval used to print the accuracy.
+    :return:
+    """
+    with Reader(os.path.join(dataset_url, 'train')) as train_reader:
+        with Reader(os.path.join(dataset_url, 'test')) as test_reader:
+            x = tf.placeholder(tf.float32, [None, 784])
+            W = tf.Variable(tf.zeros([784, 10]))
+            b = tf.Variable(tf.zeros([10]))
+            y = tf.matmul(x, W) + b
+
+            # Define loss and optimizer
+            y_ = tf.placeholder(tf.int64, [None])
+
+            # The raw formulation of cross-entropy,
+            #
+            #   tf.reduce_mean(-tf.reduce_sum(y_ * tf.log(tf.nn.softmax(y)),
+            #                                 reduction_indices=[1]))
+            #
+            # can be numerically unstable.
+            #
+            # So here we use tf.losses.sparse_softmax_cross_entropy on the raw
+            # outputs of 'y', and then average across the batch.
+            cross_entropy = tf.losses.sparse_softmax_cross_entropy(labels=y_, logits=y)
+            train_step = tf.train.GradientDescentOptimizer(0.5).minimize(cross_entropy)
+
+            train_readout = tf_tensors(train_reader)
+            train_image, train_label = transform(train_readout)
+            train_batch_image, train_batch_label = tf.train.batch(
+                [train_image, train_label], batch_size=batch_size
+            )
+
+            test_readout = tf_tensors(test_reader)
+            test_image, test_label = transform(test_readout)
+            test_batch_image, test_batch_label = tf.train.batch(
+                [test_image, test_label], batch_size=batch_size
+            )
+
+            # Train
+            print('Training model for {0} epochs with batch size {1} and log interval {2}'.format(
+                epochs, batch_size, log_interval
+            ))
+
+            with tf.Session() as sess:
+                sess.run([
+                    tf.local_variables_initializer(),
+                    tf.global_variables_initializer(),
+                ])
+                coord = tf.train.Coordinator()
+                threads = tf.train.start_queue_runners(sess=sess, coord=coord)
+                try:
+                    for i in range(epochs):
```

Comment:
Epochs is a number of full iterations over your dataset. This is not the case with your code. Note that `Reader` will never produce more then 1 epochs with this code.